include/sampling/sampling.hpp

```diff
@@ -309,54 +264,38 @@ void crhmc_sampling(PointList &randPoints,
                     HessianFunctor &h,
                     int simdLen = 1,
                     bool raw_output=false) {
-  typedef  typename Polytope::MT MatrixType;
-  typedef  crhmc_input
-          <
-                  MatrixType,
-                  Point,
-                  NegativeLogprobFunctor,
-                  NegativeGradientFunctor,
-                  HessianFunctor
-          > Input;
-  Input input = convert2crhmc_input<Input, Polytope, NegativeLogprobFunctor, NegativeGradientFunctor, HessianFunctor>(P, f, F, h);
-  typedef crhmc_problem<Point, Input> CrhmcProblem;
-  CrhmcProblem problem = CrhmcProblem(input);
-  if(problem.terminate){return;}
-  typedef typename WalkTypePolicy::template Walk
-          <
-                  Point,
-                  CrhmcProblem,
-                  RandomNumberGenerator,
-                  NegativeGradientFunctor,
-                  NegativeLogprobFunctor,
-                  Solver
-          > walk;
-  typedef typename WalkTypePolicy::template parameters
-          <
-                  NT,
-                  NegativeGradientFunctor
-          > walk_params;
-  Point p = Point(problem.center);
-  problem.options.simdLen=simdLen;
```

Comment:
This line it is deleted and not re-written. Please be sure that your changes modify only how volesti stores the points and don't affect anything else.